docs/source/about_mapstyle_vs_iterable.mdx

```diff
@@ -0,0 +1,220 @@
+# Differences between Dataset and IterableDataset
+
+## Downloading and streaming
+
+When you have a regular [`Dataset`], you can access it using `my_dataset[0]`: we have what we call "random access" to the rows.
+Such datasets are also called "map-style" datasets.
+For example you can download ImageNet-1k like this and access any row:
+
+```python
+from datasets import load_dataset
+
+imagenet = load_dataset("imagenet-1k", split="train")  # downloads the full dataset
+print(imagenet[0])
+```
+
+But one caveat is that you must have all the dataset stored on your disk or in memory, which makes it not always suitable for datasets bigger than disk.
+Because it can become inconvenient for big datasets, there exists another type of dataset: [`IterableDataset`].
+When you have an iterable dataset, you can access it using a `for` loop to load the data progressively as you iterate over the dataset.
+This way, only a small fraction of examples is loaded in memory, and you don't write anything on disk.
+
+For example, you can stream the ImageNet-1k dataset without downloading it completely:
+
+```python
+from datasets import load_dataset
+
+imagenet = load_dataset("imagenet-1k", split="train", streaming=True)  # will start loading the data when iterated over
+for example in imagenet:
+    print(example)
+    break
+```
+
+Dataset streaming can be used to read online data without writing any file on disk.
+For example you can stream datasets made out of multiple hundreds of gigabytes like [C4](https://huggingface.co/datasets/c4), [OSCAR](https://huggingface.co/datasets/oscar) or [LAION-2B](https://huggingface.co/datasets/laion/laion2B-en).
+Learn more about dataset streaming in the [Dataset Streaming Guide](./stream).
+
+This is not the only difference.
+This "lazy" behavior of iterable datasets is also present when it comes to dataset creation and processing.
```

Comment:
```suggestion
This is not the only difference though, because the "lazy" behavior of an `IterableDataset` is also present when it comes to dataset creation and processing.
```